### PR TITLE
[nnstreamer/apptest] Applying 'insmod' instead of 'modprobe'

### PR DESCRIPTION
**Pull request by ohsewon**
1. Using 'insmod' instead of 'modprobe'
2. Handling exception case that uninstalled v4l2loopback.

Signed-off-by: sewon.oh <sewon.oh@samsung.com>

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [*]Passed [ ]Failed []Skipped

### CODE REVIEW

File: ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh [hunk 1]
```diff
@@ -126,12 +123,27 @@ function pr-nnstreamer-ubuntu-apptest-run-queue() {
...
+            pushd ${REFERENCE_REPOSITORY}
+            git clone https://github.com/umlaeute/v4l2loopback.git
+            popd
+        fi        
+
         # Create virtual camera device and change authority for all.
-        sudo modprobe v4l2loopback 
+        pushd ${REFERENCE_REPOSITORY}/v4l2loopback
+        make clean && make
+        sudo insmod /lib/modules/`uname -r`/kernel/drivers/media/media.ko
```
Comment (leemgs) [inline ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh]: Please append annotation the below comment for maintenance. 

# Dependency of kernel modules: media.ko --> videodev.ko --> v4l2loopback.ko

File: ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh [hunk 2]
```diff
@@ -126,12 +123,27 @@ function pr-nnstreamer-ubuntu-apptest-run-queue() {
         echo -e "[DEBUG] USB Camera device is not enabled. It is required by {nnstreamer_example_filter|nnstreamer_example_cam}."
         echo -e "[DEBUG] Enabling virtual cam camera..." 
 
+        # Install virtual camera device kernel module, if it is not installed.
+        if [[ ! -d ${REFERENCE_REPOSITORY}/v4l2loopback ]]; then
```
Comment (leemgs) [inline ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh]: Looks good.

File: ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh [hunk 3]
```diff
@@ -126,12 +123,27 @@ function pr-nnstreamer-ubuntu-apptest-run-queue() {
         echo -e "[DEBUG] USB Camera device is not enabled. It is required by {nnstreamer_example_filter|nnstreamer_example_cam}."
         echo -e "[DEBUG] Enabling virtual cam camera..." 
 
+        # Install virtual camera device kernel module, if it is not installed.
```
Comment (leemgs) [inline ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh]: The noun is too long. 

* before: # Install virtual camera device kernel module, if it is not installed.
* after : # Install a 'v4l2loopback' kernel module to use a virtual camera device if it is not installed.

File: ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh [hunk 4]
```diff
@@ -55,6 +51,7 @@ function pr-nnstreamer-ubuntu-apptest-run-queue() {
     check_dependency wget
     check_dependency python
     check_dependency Xvnc
+    check_dependency git
```
Comment (leemgs) [inline ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh]: Looks good.

Comment (leemgs) [reply ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh]: Please append the below statement for insmod command

`check_dependency insmod`

File: ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh [hunk 5]
```diff
@@ -126,12 +123,27 @@ function pr-nnstreamer-ubuntu-apptest-run-queue() {
...
+        fi        
+
         # Create virtual camera device and change authority for all.
-        sudo modprobe v4l2loopback 
+        pushd ${REFERENCE_REPOSITORY}/v4l2loopback
+        make clean && make
+        sudo insmod /lib/modules/`uname -r`/kernel/drivers/media/media.ko
+        sudo insmod /lib/modules/`uname -r`/kernel/drivers/media/v4l2-core/videodev.ko
+        sudo insmod ./v4l2loopback.ko
         pushd /dev
```
Comment (leemgs) [inline ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh]: As I commented previously, we do not need to run "sudo chmod 777 video0" command. 
Please check it again finally in this PR. 
`[FYI] # Dependency of kernel modules: media.ko --> videodev.ko --> v4l2loopback.ko `

Comment (ohsewon) [reply ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh]: Without "sudo chmod 777 video0", element can not use '/dev/video0' cause permission denied

Comment (leemgs) [reply ci/taos/plugins-staging/pr-nnstreamer-ubuntu-apptest.sh]: > Without "sudo chmod 777 video0", element can not use '/dev/video0' cause permission denied

It's not reasonable. anyway, let's use workaroudn (e.g., chmod 777).

**Pull request merged**